### PR TITLE
PR for #18: Update package structure for compatability with Shiny

### PR DESCRIPTION
**Pull request by MosesStewart**
In this issue we . . .

- Updated the plotting functions in `main.py` and `helpers.py` to use `fig` and `ax` objects from `matplotlib.pyplot.subplots()` instead of creating plots from the `matplotlib.pyplot` class
    - Allows us to return the `fig` object, which is a requirement for displaying figures on `shiny`
- Wrote a first draft for `shiny` interface (zip file of working version attached below)
    - Can see current working version by typing `shiny run app.py` in terminal from root of directory after installing the shiny prerequisite

Adding @jmshapir
[BootstrapReport_app.zip](https://github.com/JMSLab/BootstrapReport/files/13537194/BootstrapReport_app.zip)

### CODE REVIEW

**Comment by MosesStewart** [comment]
Note: we need to pull through these changes before I can deploy online

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "BootstrapReport"
-version = "1.0.0" 
+version = "1.0.1"
```
Comment (jmshapir) [inline pyproject.toml:7]: Are the changes here backwards-compatible? If not then per [symver](https://semver.org/) we may need to make a major version change. In that case we might want to leave the version number as is for now (for development purposes) and then increment when we are ready for the next release (maybe when shiny is live.)

**Review by jmshapir — CHANGES_REQUESTED** [review]
@MosesStewart thanks!

I took an initial pass at the edits and had just one question.

On this point:
 > * Can see current working version by typing `shiny run app.py` in terminal from root of directory after installing the shiny prerequisite

is it possible for you to set up a free account at [shinyapps](https://www.shinyapps.io/) and deploy there so that we can see it in action?

File: pyproject.toml (hunk 1 above)
Comment (MosesStewart) [reply pyproject.toml]: They are not backwards-compatible. Changed back to `1.0.0` in 60b7fea!

**Comment by MosesStewart** [comment]
> is it possible for you to set up a free account at [shinyapps](https://www.shinyapps.io/) and deploy there so that we can see it in action?

@jmshapir here is what it looks like:
https://mosesstewart02.shinyapps.io/bootstrapdiagnostics1/

We won't be able to see the plots until the repository is updated. Then I'll have to make a new website to deploy.

**Review by jmshapir — APPROVED** [review]
@MosesStewart looks great!

Clear to proceed from my standpoint.

For the shiny app a few things (mostly from our call):
* We can store the shiny code in `BootstrapReport` in a new directory called `/app`.
* If not difficult to implement, it would be nice to have behavior along the lines of:
  * If user specifies name of column with replicates, use that column
  * If user does not specify a name, use first data column regardless of its name
* It would be great if the app can have some text at the bottom that directs users to the `BootstrapReport` repo for documentation and other details. That way the app is fully self-contained and we don't have to duplicate information.

**Pull request merged**